docs/dev/connectors/pravega.md

```diff
@@ -0,0 +1,398 @@
+---
+title: "Pravega Connector"
+nav-title: Pravega
+nav-parent_id: connectors
+nav-pos: 6
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+## Pravega Introduction
+[Pravega](http://pravega.io/) is an open-source, high-performance, durable, elastic and strongly-consistent
+stream storage for continuous and unbounded data.
+
+Here are some need-to-know Pravega terminologies.
+- **Stream**  A core abstraction -- `Stream` is provided to store unbounded events which aligns the design of Flink's DataStream API.
+- **Scope**  Each `Stream` is under a namespace/tenant referred as `Scope`.
+- **Segment**  Each `Stream` is split into a set of shards or partitions generally referred as `Segments` which can be auto-scaled.
+- **Routing Key**  Every event written to a Pravega `Stream` has an associated `Routing Key` in an append-only way into a `Segment`.
+A `Routing Key` is a string used by developers to group similar events which is also the basis for event ordering guarantee.
+- **Streamcut**  A `StreamCut` represents a specific position in a Pravega `Stream`, which may be obtained from various API interactions with the Pravega client.
+
+## Pravega connector
+Pravega connector can be used to build end-to-end real-time stream processing pipelines
+(see [Samples](https://github.com/pravega/pravega-samples)) that use Pravega as the stream storage/message bus
+and Apache Flink for computing engine over the streams.
+
+### Features & Highlights
+- Exactly-once processing guarantees for both Reader and Writer, supporting end-to-end exactly-once processing pipelines
+- Seamless integration with Flink's checkpoints and savepoints.
+- Parallel Readers and Writers supporting high throughput and low latency processing.
+- Table API support to access Pravega Streams for both Batch and Streaming use case.
+
+### Dependency
+To use this connector, please pick the proper dependency and add it to your project:
```

Comment:
Let's make explicit references to maven and gradlew, and adapt the samples below accordingly:

`To use the connector, add the appropriate dependency to your project via Maven or Gradlew`.

Reply:
I think as a Flink document, it should follow the Flink rule. Flink document only reference maven format for all its dependencies.
(e.g. https://ci.apache.org/projects/flink/flink-docs-stable/dev/connectors/kafka.html#usage 
https://ci.apache.org/projects/flink/flink-docs-release-1.9/getting-started/tutorials/datastream_api.html) 

There is not a template in the doc for switch in maven and other compiling tools, so I can only adopt your idea to keep showing these two together on the page. Your suggestion makes sense but I guess the Flink community wants to keep the tradition among the docs.

Reply:
> There is not a template in the doc for switch in maven and other compiling tools, so I can only adopt your idea to keep showing these two together on the page. Your suggestion makes sense but I guess the Flink community wants to keep the tradition among the docs.

Here is a Flink quickstart document that shows both Maven and Gradle configs: https://ci.apache.org/projects/flink/flink-docs-stable/dev/projectsetup/java_api_quickstart.html#gradle

If it is just as short example, I think it is safe and helpful to include Gradle configuration stanzas here as well.

Reply:
Makes sense, I'll add the `gradle` section.